### PR TITLE
[Merged by Bors] - feat: port Algebra.CharZero.Quotient

### PR DESCRIPTION
**Pull request by mo271**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/Algebra/CharZero/Quotient.lean [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+
+end AddSubgroup
+
+namespace quotientAddGroup
+
+theorem zmultiples_zsmul_eq_zsmul_iff {ψ θ : R ⧸ AddSubgroup.zmultiples p} {z : ℤ} (hz : z ≠ 0) :
+    z • ψ = z • θ ↔ ∃ k : Fin z.natAbs, ψ = θ + (k : ℕ) • (p / z : R) := by
+  induction ψ using Quotient.inductionOn'
+  induction θ using Quotient.inductionOn'
+  have : (Quotient.mk'' : R → R ⧸ AddSubgroup.zmultiples p) = coe := rfl
```
Comment (mo271) [inline Mathlib/Algebra/CharZero/Quotient.lean:64]: Currently stuck on how to port the `coe` here.

Comment (LukasMias) [reply Mathlib/Algebra/CharZero/Quotient.lean:64]: Replacing `coe` by the usual arrow shenanigans works, so `((↑) : R → R ⧸ Zp)`. Together with some mild fixes to some other proofs, that's the file done as far as I can tell!

Comment (mo271) [reply Mathlib/Algebra/CharZero/Quotient.lean]: perfect, thanks!

**Review by dupuisf — COMMENTED** [review]
Thanks!

bors r+

**Pull request closed without merging**